File: app/frontend/.claude/hooks/load-skill.py
#!/usr/bin/env python
"""Auto-load project skills when this subdirectory is the project root.

- NATIVE_SKILL is always loaded (once per session) on SessionStart / Edit / Write.
- EXTRA_MAP adds conditional skills that load only when the Edit|Write target
  matches a given path token (e.g. editing infra/env/*.env triggers env-sync).

Each skill is injected at most once per session via a sentinel file.
"""
from __future__ import annotations

import json
import sys
from pathlib import Path

NATIVE_SKILL = "frontend-storybook-architecture"

EXTRA_MAP = [
    ("/infra/env/", "env-sync"),
]


def main() -> int:
    try:
        payload = json.load(sys.stdin)
    except Exception:
        payload = {}

    session_id = payload.get("session_id") or "default"
    event_name = payload.get("hook_event_name") or "SessionStart"
    file_path = (payload.get("tool_input") or {}).get("file_path") or ""

    hook_dir = Path(__file__).resolve().parent
    repo_root = hook_dir.parents[3]

    sentinel_dir = hook_dir.parent / "cache" / "skill-loaded" / session_id
    sentinel_dir.mkdir(parents=True, exist_ok=True)

    # Determine which skills to inject this invocation.
    candidates: list[str] = []

    # Native skill: always candidate (sentinel check below).
    candidates.append(NATIVE_SKILL)

    # Conditional skills: only when file_path matches an EXTRA_MAP token.
    if file_path:
        norm = file_path.replace("\\", "/")
        for token, skill in EXTRA_MAP:
            if token in norm:
                candidates.append(skill)

    # Resolve to actual injections (skip if already loaded this session).
    parts: list[str] = []
    for skill in candidates:
        sentinel = sentinel_dir / skill
        if sentinel.exists():
            continue
        skill_file = repo_root / ".agents" / "skills" / skill / "SKILL.md"
        if not skill_file.is_file():
            continue
        sentinel.touch()
        content = skill_file.read_text(encoding="utf-8")
        rel = f".agents/skills/{skill}/SKILL.md"
        parts.append(
            f"Auto-loaded project skill from {rel} (frontend workspace):\n\n{content}"
        )

    if not parts:
        return 0

    out = {
        "hookSpecificOutput": {
            "hookEventName": event_name,
            "additionalContext": "\n\n---\n\n".join(parts),
        }
    }
    json.dump(out, sys.stdout)
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: app/frontend/.claude/hooks/load-skill.js
#!/usr/bin/env node
// Auto-load project skills when this subdirectory is the project root.
// Ported from load-skill.py to avoid Windows Python App Execution Alias stub issues.
//
// - NATIVE_SKILL is always loaded (once per session) on SessionStart / Edit / Write.
// - EXTRA_MAP adds conditional skills that load only when the Edit|Write target
//   matches a given path token (e.g. editing infra/env/*.env triggers env-sync).
//
// Each skill is injected at most once per session via a sentinel file.

const fs = require('fs');
const path = require('path');

const NATIVE_SKILL = 'frontend-storybook-architecture';

const EXTRA_MAP = [
  ['/infra/env/', 'env-sync'],
];

const WORKSPACE = 'frontend';

function main() {
  let raw;
  try {
    raw = fs.readFileSync(0, 'utf8');
  } catch {
    return 0;
  }

  let payload;
  try {
    payload = JSON.parse(raw);
  } catch {
    payload = {};
  }

  const sessionId = payload.session_id || 'default';
  const eventName = payload.hook_event_name || 'SessionStart';
  const filePath = (payload.tool_input && payload.tool_input.file_path) || '';

  // repo_root = 4 levels up from hook file.
  // .claude/hooks/load-skill.js → .claude/hooks → .claude → <workspace> → app → repo_root.
  const hookDir = __dirname;
  const repoRoot = path.resolve(hookDir, '..', '..', '..', '..');

  const sentinelDir = path.resolve(hookDir, '..', 'cache', 'skill-loaded', sessionId);
  fs.mkdirSync(sentinelDir, { recursive: true });

  const candidates = [];

  if (NATIVE_SKILL) {
    candidates.push(NATIVE_SKILL);
  }

  if (filePath) {
    const norm = filePath.replace(/\\/g, '/');
    for (const [token, skill] of EXTRA_MAP) {
      if (norm.includes(token)) candidates.push(skill);
    }
  }

  const parts = [];
  for (const skill of candidates) {
    const sentinel = path.join(sentinelDir, skill);
    if (fs.existsSync(sentinel)) continue;
    const skillFile = path.join(repoRoot, '.agents', 'skills', skill, 'SKILL.md');
    if (!fs.existsSync(skillFile) || !fs.statSync(skillFile).isFile()) continue;
    fs.writeFileSync(sentinel, '');
    const content = fs.readFileSync(skillFile, 'utf8');
    const rel = `.agents/skills/${skill}/SKILL.md`;
    parts.push(
      `Auto-loaded project skill from ${rel} (${WORKSPACE} workspace):\n\n${content}`
    );
  }

  if (parts.length === 0) return 0;

  const out = {
    hookSpecificOutput: {
      hookEventName: eventName,
      additionalContext: parts.join('\n\n---\n\n'),
    },
  };
  process.stdout.write(JSON.stringify(out));
  return 0;
}

process.exit(main());
